packages/graphql-transformer-common/src/FunctionResourceIDs.ts

```diff
@@ -9,6 +10,13 @@ export class FunctionResourceIDs {
     return `${FunctionResourceIDs.FunctionDataSourceID(name, region)}Role`;
   }
 
+  static FunctionIAMRoleName(name: string, withEnv: boolean = false): string {
+    if (withEnv) {
+      return `${simplifyName(name).slice(0, 18)}${shortid.generate()}`;
```

Comment:
We don't want a new UUID each time we do this. This would result in deleting an IAM role and then creating a new IAM role each time. Use the first 4 chars of the AppId instead.

Reply:
Updated to use a hash of the name added at the end.